.github/workflows/unit-tests.yml

```diff
@@ -346,3 +346,62 @@ jobs:
 
       - name: Run Tests
         uses: ./.github/actions/run-tests
+
+  emscripten:
+    # Note: the Python version, Emscripten toolchain version are determined
+    # by the Pyodide version. The appropriate versions can be found in the
+    # Pyodide repodata.json "info" field, or in the Makefile.envs file:
+    # https://github.com/pyodide/pyodide/blob/stable/Makefile.envs#L2
+    # The Node.js version can be determined via Pyodide:
+    # https://pyodide.org/en/stable/usage/index.html#node-js
+    name: Pyodide build
+    runs-on: ubuntu-22.04
+    concurrency:
+      # https://github.community/t/concurrecy-not-work-for-push/183068/7
+      group: ${{ github.event_name == 'push' && github.run_number || github.ref }}-wasm
+      cancel-in-progress: true
+    steps:
+      - name: Checkout pandas Repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python for Pyodide
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11.3'
+
+      - name: Set up Emscripten toolchain
+        uses: mymindstorm/setup-emsdk@v14
+        with:
+          version: '3.1.46'
+          actions-cache-folder: emsdk-cache
+
+      - name: Install pyodide-build
+        run: pip install "pydantic<2" "pyodide-build==0.25.1"
+
+      - name: Build pandas for Pyodide
+        run: |
+          pyodide build
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      - name: Set up Pyodide virtual environment
+        run: |
+          pyodide venv .venv-pyodide
+          source .venv-pyodide/bin/activate
+          pip install dist/*.whl
+
+      - name: Test pandas for Pyodide
+        env:
+          PANDAS_CI: 1
+        run: |
+          source .venv-pyodide/bin/activate
+          pip install "pytest<8.1.0" hypothesis
```

Comment:
Curious why pytest needs pinning?

Reply:
I have been using it for historical reasons (since the NumPy job earlier this year kept the pin, and no one bothered to remove it) – it used to be pinned earlier (https://github.com/pyodide/pyodide/issues/3978) but I believe the fix has been backported and is generally available with our current version (i.e., v0.25.1). Let me remove the pin in another commit.

Reply:
Addressed in 7f4715f49b7011de2430f73dadb2acbd38b9421f.

Reply:
Oops, I misread this. I thought your comment said `pydantic`, but this was `pytest`. I don't think it needs a pin, let me remove it. I deleted my earlier comments.

Reply:
It could be unpinned. Back when this PR was opened, `pytest` had inadvertently pushed out a breaking change for many users which was later yanked, I think. Let me try to remove the pin in another commit.

Reply:
Un-pinned in ab911d1b32bca21b48872024d60dae54d5844be5, and the test suite is proceeding to run the tests after their collection.